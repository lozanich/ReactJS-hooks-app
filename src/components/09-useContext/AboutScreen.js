import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);
  const handleLogout = () => {
    setUser({});
  };
  return (
    <div>
      <h1>About Screen</h1>
      <hr />

      <pre>{JSON.stringify(user)}</pre>

      <button onClick={handleLogout} className="btn btn-warning">
        Logout
      </button>
    </div>
  );
};
