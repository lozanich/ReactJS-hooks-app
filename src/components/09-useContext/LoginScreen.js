import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);
  const user = {
    id: 123838,
    name: "Ivan Lozano",
    email: "lozanich598@gmail.com",
  };
  return (
    <div>
      <h1>Login Screen</h1>
      <hr />

      <button onClick={() => setUser(user)} className="btn btn-primary">
        Login
      </button>
    </div>
  );
};
