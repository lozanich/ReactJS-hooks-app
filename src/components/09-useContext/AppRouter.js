import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { Navbar } from "./Navbar";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Navbar />

        <div className="container">
          <Switch>
            <Route exact path="/" component={HomeScreen}></Route>
            <Route exact path="/about" component={AboutScreen}></Route>
            <Route exact path="/login" component={LoginScreen}></Route>

            <Route component={HomeScreen}></Route>
          </Switch>
        </div>
      </div>
    </Router>
  );
};
