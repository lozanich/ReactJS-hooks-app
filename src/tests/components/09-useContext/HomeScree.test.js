import React from "react";
import "@testing-library/jest-dom";
import { mount, shallow } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Pruebas en <HomeScreen />", () => {
  const user = {
    name: "Ivan Lozano",
    email: "lozanich598@gmail.com",
  };
  const wrapper = mount(
    <UserContext.Provider
      value={{
        user,
      }}
    >
      <HomeScreen />
    </UserContext.Provider>
  );

  test("Debe de coincidir con el screenShot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
