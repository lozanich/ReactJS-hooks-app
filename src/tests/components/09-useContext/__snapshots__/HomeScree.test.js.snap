// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> Debe de coincidir con el screenShot 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "Ivan Lozano",
   "email": "lozanich598@gmail.com"
}
  </pre>
</div>
`;
